client/src/components/Pages/MainPage.js

```diff
@@ -37,7 +37,11 @@ export default function MainPage() {
   }
   // filtering by seach category
   if (searchCategory) {
```

Comment:
It's not the best approach.
I think it's better to add `categoryId` to the `records`.
I've just done at API.
Then you can filter by `categoryId`.